src/realm/object-store/sync/sync_session.cpp

```diff
@@ -335,10 +335,18 @@ void SyncSession::download_fresh_realm(sync::ProtocolErrorInfo::Action server_re
                 server_requests_action);
         }
     }
+
+    std::vector<char> encryption_key;
+    {
+        util::CheckedLockGuard lock(m_config_mutex);
+        encryption_key = m_config.encryption_key;
+    }
+
     DBOptions options;
-    options.encryption_key = m_db->get_encryption_key();
     options.allow_file_format_upgrade = false;
     options.enable_async_writes = false;
+    if (!encryption_key.empty())
+        options.encryption_key = encryption_key.data();
```

Comment:
Does the encryption key need to live longer than this function, or is it only needed to open the realm initially?

Reply:
It's only used when opening the file. The encryption layer constructs a context using the key, and explicitly copies the key when the underlying thing doesn't do that already.

Given that they don't reliably work, `File::get_encryption_key()` and `DB::get_encryption_key()` probably shouldn't exist.